pkg/interface.go

```diff
@@ -38,6 +38,10 @@ type Client interface {
 	// GenerateVendorCACSR generates a Certificate Signing Request (CSR) for a Vendor CA Certificate for the current vendor.
 	// If successful it returns the CSR as DER-encoded PKCS10.
 	GenerateVendorCACSR(name string) ([]byte, error)
+	// StoreCertificate stores the given certificate for the given key. If there's a private key present, it must match
+	// the public key (otherwise an error is returned). If there's an existing certificate entry, it will be overridden.
+	// This is typically used to store a certificate resulting from the CSR generated by GenerateVendorCACSR.
+	StoreCertificate(key types.KeyIdentifier, certificate *x509.Certificate) error
```

Comment:
Overriding might be a problem for a vendor CA certificate? If it's used but not yet distributed? Is a graceful transition period needed?

Reply:
I don't like this function anyways. It's input is `GenerateVendorCACSR()` -> (send to CA, get certificate in return) -> `StoreCertificate()`.

First one is high level, business/process oriented, latter one is more like a repository method. Maybe I should change it to `RegisterVendorCACertificate()`.

Used but not yet distributed can be a problem in any case. Maybe the Network Authority should renew certificates with 1 day in future (`not before = today() + 1 day`)

Reply:
Made it more specific as the counterpart of `GenerateVendorCACSR`: `StoreVendorCACertificate`. We can move it later to a more high level layer.

Regarding certificates used but not yet distributed; they're distributed by registering them on the network so we can't say with confidence when all other nodes have the new certificate (since it's eventual consistent). Best bet is to use the fact that it takes some time after the Network Authority issued and registered a certificate to send it back to the vendor and for the vendor to load it into their node?